File: app/src/at-tamle/java/asiantech/internship/winter/baitap/bai30.java
package asiantech.internship.winter.baitap;
import java.util.*;

// ho dem ten > ten dem ho
public class bai30 {
	 public static void chuyendoi(String s){
	        StringTokenizer tg= new StringTokenizer(s);
	        StringBuffer kq= new StringBuffer();
	        String s1[]= new String[10];
	        int i=1;
	        while(tg.hasMoreTokens()){
	            s1[i]= tg.nextToken();
	            i++;


	        }
	        kq.append(s1[3]).append("  "); kq.append(s1[1]).append("  "); kq.append(s1[2]).append("  ");
	       System.out.println(kq);
	    }
	    public static void main(String[] args) {
	        String s0= new String();
	        Scanner in= new Scanner(System.in);
	        System.out.println("nhap xau ho ten: ");
	        s0= in.nextLine();
	        chuyendoi(s0);

	  }

}

File: app/src/at-trinhnguyen/java/asiantech/internship/winter/javacore/Bai30.kt
package com.congtrinh

import java.util.*

/*
Bài 30.Viết chương trình thực hiện nhập một xâu họ tên theo cấu trúc:
họ...đệm...tên; chuyển xâu đó sang biểu diễn theo cấu trúc tên…họ…đệm.
 */
object Bai30 {
    @JvmStatic
    fun main(args: Array<String>) {
        val scanner = Scanner(System.`in`)
        print("Enter string: ")
        val str = scanner.nextLine()
        println(flipLastAndFirstName(str))
    }

    private fun firstCharToUpperCase(str: String): String {
        val s: String = str.substring(0, 1)
        val strOutput: String
        strOutput = str.replaceFirst(s.toRegex(), s.toUpperCase())
        return strOutput
    }

    private fun standardized(strInput: String): String {
        var strOutput = ""
        val strToken = StringTokenizer(strInput, " ,\t,\r")
        strOutput += "" + firstCharToUpperCase(strToken.nextToken())
        while (strToken.hasMoreTokens()) {
            strOutput += " " + firstCharToUpperCase(strToken.nextToken())
        }
        return strOutput
    }

    private fun flipLastAndFirstName(strInput:String):String{
        val str= standardized(strInput)
        val strToken=StringTokenizer(str," ")
        var lastName=""
        val firstName: String
        for(i in 0 until (strToken.countTokens()-1)){
            lastName+=" "+strToken.nextToken()
        }
        firstName=strToken.nextToken()
        return firstName+lastName
    }
}